### PR TITLE
Replace docker-compose with docker compose in just scripts and docs

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #3876 by @zackkrida

## Description

This PR only replaces `docker-compose` in `justfile`s and documentation with `docker compose`.
It does not simplify the `docker ps` script.
@zackkrida mentions the replacement of underscores with dashes for container names in the issue: should this be done manually? Would it also affect production? 

I think it's best to keep the scope of this PR small to fix the critical issue of CI failures due to `docker-compose` command not being found.

## Testing Instructions

The CI should pass.
All of the `just` commands that used to use `docker-compose` should work as usual.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: ingestion_server/test/integration_test.py [hunk 1]
```diff
@@ -166,10 +166,11 @@ def _load_data(conn, table_names):
 
 
 def _compose_cmd(compose_path, cmd: list[str], **kwargs):
-    """Run a Docker Compose command"""
+    """Run a Docker compose command"""
```
Comment (dhruvkb) [inline ingestion_server/test/integration_test.py:169]: This was okay as it was.

File: documentation/frontend/reference/playwright_tests.md [hunk 2]
```diff
@@ -31,8 +31,8 @@ cross-platform browser differences from creating flaky test behavior. We run
 both end-to-end and visual-regression tests in the same container to save from
 having to run the Nuxt production build twice.
 
-Having docker and docker-compose is a pre-requisite to running the playwright
-tests locally. Please follow
+Having docker and Compose V2 is a pre-requisite to running the playwright tests
```
Comment (dhruvkb) [inline documentation/frontend/reference/playwright_tests.md:34]:
```suggestion
Having Docker and Compose V2 is a pre-requisite to running the Playwright tests
```

File: docker-compose.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-version: "2.4"
+version: "3.4"
```
Comment (dhruvkb) [inline docker-compose.yml:1]: This is a big change. I'm not sure we should be switching to v3 without a thorough audit of the effects. Afaik, v2 is still supported with Compose v2 so I would prefer to not change this.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
The changes look good to me.

The underscore and dash change is a non-issue for us especially if we use the `docker compose exec` command with service names instead of container names.

I have more confidence in the changes because many folks on the team, myself included, have been using Compose v2 for a while and it has not affected anything for us.

I do have some nits about capitalisation but those are not blockers. The blocker is the v2 &rarr; v3 upgrade in `docker-compose.yml`.

File: docker-compose.yml (hunk 3 above)
Comment (obulat) [reply docker-compose.yml:1]: The comment in the same file links to documentation for v3 of compose files, so I decided to try fixing this. The JetBrains IDEs usually complain about some parts of this file (mainly, the extension fields), but when you update the version to 3.4, the red disappears. However, I re-checked and _extension fields_ were introduced in version 2.4[^1], so it's confusing.

I'll revert this change.
[^1]: https://docs.docker.com/compose/compose-file/compose-versioning/#version-24

Comment (obulat) [inline docker-compose.yml:1]:
```suggestion
version: "2.4"
```

Comment (obulat) [reply docker-compose.yml:1]: Revert the update of docker-compose.yml version

File: ingestion_server/test/integration_test.py (hunk 1 above)
Comment (obulat) [inline ingestion_server/test/integration_test.py:169]:
```suggestion
    """Run a Docker Compose command"""
```

File: docker-compose.yml (hunk 3 above)
Comment (dhruvkb) [reply docker-compose.yml]: IIRC there was some issues with the resource limits when using version 3.x. But we can eventually switch up once we verify things are working normally. Can you create a an issue to do the audit and upgrade?

**Review by dhruvkb — APPROVED** [review]

File: docker-compose.yml (hunk 3 above)
Comment (obulat) [reply docker-compose.yml]: Done: #4020

**Review by AetherUnbound — APPROVED** [review]
Thanks for making this change! I encountered it yesterday but it was successful after a second run. Weird!

**Pull request merged**